app/models/client_opportunity_match.rb

```diff
@@ -377,6 +382,33 @@ def self.text_search(text)
     )
   end
 
+  def self.send_summary_emails
+    config = Config.last
+    return if config.never_send_match_summary_email?
+    return unless config.send_match_summary_email_on == Date.current.wday
+
+    match_groups = [
+      # stalled
+      stalled.preload(:contacts),
+      # canceled
+      canceled.where(updated_at: 1.week.ago..).preload(:contacts),
+      # expiring
+      where(shelter_expiration: Date.current..Date.current + 1.week).preload(:contacts),
+      # expired
+      expired.where(shelter_expiration: 1.week.ago..).preload(:contacts),
+      # active
+      active.preload(:contacts),
+    ]
+
+    contact_ids = match_groups.map do |group|
+      group.map { |d| d.contacts.map(&:id) }.flatten
```

Comment:
I think this is the same:
```suggestion
      group.flat_map { |d| d.contacts.map(&:id) }
```